xc7/zynq7.cmake

```diff
@@ -0,0 +1,78 @@
+include(make/project_xray.cmake)
+
+add_subdirectory(primitives)
+
+# Convert prjxray database into VPR channels, pin definitions for tile types,
+# and create direct connection list for arch.xml.
+project_xray_prepare_database(
+  PART zynq7
+  )
+
+set(ROI_PART xc7z010clg400-1)
+set(ROI_DIR ${PRJXRAY_DB_DIR}/zynq7/harness/zybo/swbut)
+
+add_subdirectory(tiles)
+add_subdirectory(blocks)
+
+
+define_arch(
```

Comment:
Given that most of this is the same, should make a function to abstract the stuff that is the "same".  For example, the YOSYS_SCRIPT line is going to change for BRAM support, and both artix7 and zync7 should likely use the same YOSYS_SCRIPT line.